drivers/hmis/app/graphql/types/base_audit_event.rb

```diff
@@ -34,6 +34,16 @@ def self.build(node_class, field_permissions: nil, excluded_keys: nil, transform
           # Check if user has permission to view audit history for this particular field (for example SSN/DOB on Client)
           current_user.permissions_for?(record, *Array.wrap(field_permissions[key]))
         end
+
+        define_method(:authorize_object) do |item_type, record|
```

Comment:
This enables us to hide all fields on a given item type, rather than making the determination key-by-key. Previously, we only needed key-by-key masking because the only keys we masked out were DOB and SSN. Now that we are masking out contact info, this is coming from the generic `CustomClientContactPoint` object, and has keys like `system` and `value` -- and I felt nervous that there might be overlap with keys in some other object that we would not want to mask. (This also enables the removal of the `CustomClientName` fields from `FIELD_PERMISSIONS_FOR_AUDIT` in drivers/hmis/app/graphql/types/hmis_schema/client.rb.) Please let me know if this addition looks ok!